wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -794,7 +806,40 @@ public void waitForData() {
    * @return true if there is new data, otherwise false
    */
   public boolean waitForData(double timeout) {
-    return HAL.waitForDSDataTimeout(timeout);
+    long startTime = RobotController.getFPGATime();
+    long timeoutMicros = (long) (timeout * 1000000);
+    m_waitForDataMutex.lock();
+    try {
+      try {
```

Comment:
I don't think this inner try is necessary. Wouldn't bumping this up one level work fine?